servers/physics_2d/broad_phase_2d_bvh.h

```diff
@@ -37,7 +37,50 @@
 #include "core/math/vector2.h"
 
 class BroadPhase2DBVH : public BroadPhase2DSW {
-	BVH_Manager<CollisionObject2DSW, true, 128, Rect2, Vector2> bvh;
+	template <class T>
+	class UserPairTestFunction {
+	public:
+		static bool user_pair_check(const T *p_a, const T *p_b) {
+			// return false if no collision, decided by masks etc
+			return p_a->test_collision_mask(p_b);
+		}
+	};
+
+	template <class T>
+	class UserCullTestFunction {
+		static bool _cull_pairing_mask_test_hit(uint32_t p_maskA, uint32_t p_typeA, uint32_t p_maskB, uint32_t p_typeB) {
+			bool A_match_B = p_maskA & p_typeB;
+
+			if (!A_match_B) {
+				bool B_match_A = p_maskB & p_typeA;
+				if (!B_match_A) {
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+	public:
+		static bool user_cull_check(const T *p_a, const T *p_b) {
```

Comment:
If we were filtering objects out using tree mask it seems we could also get rid of the extra callback here. I might be missing something in the way the BVH trees work though so let me know if it's the case :)

Reply:
If we use the tree masks for this, we would just make this return true, and it should compile out. This check is still used in the render trees so we can't remove it altogether.

Reply:
This now just returns true. I could remove this code and just use the Dummy version specified in BVH_tree, or leave this here as a stub in case we want to use it in future. Either is fine, whichever is clearer.